### PR TITLE
chore(composer): minimal PHP version is now 8.1

### PR DESCRIPTION
**Pull request by jdalsem**
Also we now fully support PHP 8.3

### CODE REVIEW

**Review by jeabakker — APPROVED** [review]

File: engine/tests/phpunit/unit/Elgg/RouterUnitTest.php [hunk 1]
```diff
@@ -1081,7 +1048,7 @@ public function testCanRespondToAjax2ViewRequestForARegisteredViewWhichForwards(
 		], ELGG_JSON_ENCODING);
 
 		$this->assertEquals($output, $response->getContent());
-
+
```
Comment (jeabakker) [inline engine/tests/phpunit/unit/Elgg/RouterUnitTest.php:1051]: ????

File: engine/tests/phpunit/unit/Elgg/Users/AccountsServiceUnitTest.php [hunk 2]
```diff
@@ -12,7 +12,7 @@ class AccountsServiceUnitTest extends UnitTestCase {
 	/**
 	 * @var int minimal username length during testing
 	 */
-	protected $minusername = 6;
+	protected const minusername = 6;
```
Comment (jeabakker) [inline engine/tests/phpunit/unit/Elgg/Users/AccountsServiceUnitTest.php:15]: constants should be uppercase

**Review by jeabakker — CHANGES_REQUESTED** [review]

File: engine/tests/classes/Elgg/BaseTestCase.php [hunk 3]
```diff
@@ -189,6 +190,15 @@ protected function tearDown(): void {
 			_elgg_services()->system_messages->dumpRegister();
 		}
 		
+		if (CompositeCache::isMemcacheAvailable()) {
```
[the remark this answers is not in the document]
Comment (jdalsem) [reply engine/tests/classes/Elgg/BaseTestCase.php]: talked about it, not going to check it

File: engine/tests/classes/Elgg/BaseTestCase.php [hunk 4]
```diff
@@ -189,6 +189,17 @@ protected function tearDown(): void {
 			_elgg_services()->system_messages->dumpRegister();
 		}
 		
+		// performing some additional cleanup as we can't rely on php garbage collection
+		foreach (CacheManager::getInstances() as $instance ){
```
Comment (jeabakker) [inline engine/tests/classes/Elgg/BaseTestCase.php:193]: `$instance ){` => `$instance) {`

**Review by jeabakker — CHANGES_REQUESTED** [review]

**Review by jeabakker — APPROVED** [review]

**Pull request merged**